public/app/features/transformers/editors/CalculateFieldTransformerEditor.tsx

```diff
@@ -356,6 +360,78 @@ export class CalculateFieldTransformerEditor extends React.PureComponent<
     );
   }
 
+  //---------------------------------------------------------
+  // Unary Operator
+  //---------------------------------------------------------
+
+  updateUnaryOptions = (v: UnaryOptions) => {
+    const { options, onChange } = this.props;
+    onChange({
+      ...options,
+      mode: CalculateFieldMode.UnaryOperation,
+      unary: v,
+    });
+  };
+
+  onUnaryOperationChanged = (v: SelectableValue<string>) => {
+    const { unary } = this.props.options;
+    this.updateUnaryOptions({
+      ...unary!,
+      operator: v.value! as UnaryOperationID,
+    });
+  };
+
+  onUnaryValueChanged = (v: SelectableValue<string>) => {
+    const { unary } = this.props.options;
+    this.updateUnaryOptions({
+      ...unary!,
+      fieldName: v.value!,
+    });
+  };
+
+  renderUnaryOperation(options?: UnaryOptions) {
+    options = defaults(options, { operator: UnaryOperationID.Abs });
+
+    let found = !options?.fieldName;
+    const names = this.state.names.map((v) => {
+      if (v === options?.fieldName) {
+        found = true;
+      }
+      return { label: v, value: v };
+    });
+
+    const ops = unaryOperators.list().map((v) => {
+      return { label: v.id, value: v.id };
+    });
+
+    const fieldName = found ? names : [...names, { label: options?.fieldName, value: options?.fieldName }];
+
+    return (
+      <div className="gf-form-inline">
+        <div className="gf-form">
+          <div className="gf-form-label width-8">Operation</div>
+        </div>
+        <div className="gf-form">
+          <Select
+            className="width-8 gf-form-spacing"
+            options={ops}
+            value={options.operator ?? ops[0].value}
+            onChange={this.onUnaryOperationChanged}
+          />
+          <div className="gf-form-label width-1">(</div>
+          <Select
+            placeholder="Field"
+            className="min-width-7 gf-form-spacing"
+            options={fieldName}
+            value={options?.fieldName}
+            onChange={this.onUnaryValueChanged}
+          />
+          <div className="gf-form-label width-1">)</div>
+        </div>
+      </div>
```

Comment:
I would prefer if this was using the new form elements. Not sure how consistent it would look with the form for the binary operators, maybe it's not that big of a deal? Maybe it's time to change the binary form the the new form elements as well?

Reply:
I agree! I'll look into that!

Reply:
Modified the editor to now use new ui elements!